Modules/_posixsubprocess.c

```diff
@@ -677,6 +705,47 @@ subprocess_fork_exec(PyObject* self, PyObject *args)
         cwd_obj2 = NULL;
     }
 
+    if (groups_list != Py_None) {
+        Py_ssize_t i;
+        int gid;
+
+        if (!PyList_Check(groups_list)) {
+            PyErr_SetString(PyExc_TypeError, "setgroups argument must be a sequence");
+            goto cleanup;
+        }
+        num_groups = PySequence_Size(groups_list);
+
+        if (num_groups < 0)
+            goto cleanup;
+
+        if (num_groups > MAX_GROUPS) {
+            PyErr_SetString(PyExc_ValueError, "too many groups");
+            goto cleanup;
+        }
+
+        for(i = 0; i < num_groups; i++) {
+            PyObject *elem;
+            elem = PySequence_GetItem(groups_list, i);
+            if (!elem)
+                goto cleanup;
+            if (!PyLong_Check(elem)) {
+                PyErr_SetString(PyExc_TypeError,
+                                "groups must be integers");
+                Py_DECREF(elem);
+                goto cleanup;
+            } else {
+                gid = PyLong_AsUnsignedLong(elem);
```

Comment:
is UnsignedLong sufficient for gid_t on all platforms?  (be consistent with whatever posixmodule.c does for it's  setgid implementation bounds checking - if this is already sufficient, great, just add a comment here stating why :)

same

Reply:
posixmodule.c has a fairly complicated function called `_Py_Gid_Converter` that tries to find store it a signed long before falling back on storing it as an UnsignedLong if it doesn't fit in a Long (it also does bounds checking on the value). Since we are doing the bounds checking on the Python side, I figured just directly putting it in an UnsignedLong is probably the easiest way.

Reply:
Would this comment be a reasonable explaination, or should I try to avoid referring to other modules?
```c
                 /* In posixmodule.c UnsignedLong is used as a fallback value
                  * if the value provided does not fit in a Long. Since we are
                  * already doing the bounds checking on the Python side, we
                  * can go directly to an UnsignedLong here. */
```

Reply:
yeah that comment makes sense.  life is easier in this module for a few things like this as python code calling it gets to be the pedantic part.